### PR TITLE
remove check under linux for ~/.matplotlib

### PR DESCRIPTION
**Pull request by nbrunett**
Here's a shot at removing any checks for, or use of, ~/.matplotlib under linux. Motivated by discussion in issue #6885. Mainly just got rid of the check in \__init\__.py and some docstring stuff there. Also took a crack at updating the environment variable FAQ since it made reference to this stuff. Passed same tests as matplotlib/matplotlib/master did when running matplotlib.test().

@tacaswell mentioned having this for 2.1 but I'm not sure how to target a particular release. I'd be happy to try again if there's a way to do that explicitly on my end.

### CODE REVIEW

**Comment by tacaswell** [comment]
Can you rebase to remove the merge from upstream master into your feature branch?

**Comment by tacaswell** [comment]
Otherwise looks good.

The master branch will be the 2.1 release so this is the correct place to open the pull request.

File: doc/faq/environment_variables_faq.rst [hunk 1]
```diff
@@ -25,10 +25,11 @@ Environment Variables
 
   This is the directory used to store user customizations to matplotlib, as
   well as some caches to improve performance. If :envvar:`MPLCONFIGDIR` is not
-  defined, :file:`{HOME}/.matplotlib` is used if it is writable.
-  Otherwise, the python standard library :func:`tempfile.gettmpdir` is
-  used to find a base directory in which the :file:`matplotlib`
-  subdirectory is created.
+  defined, :file:`{HOME}/.config/matplotlib` is generally used on unix-like
+  systems and :file:`{HOME}/.matplotlib` is used on other platforms, if they are
+  writable. Otherwise, the python standard library :func:`tempfile.gettmpdir`
```
Comment (QuLogic) [inline doc/faq/environment_variables_faq.rst]: Should say `gettempdir`.

**Comment by nbrunett** [comment]
I gave the rebase a try. It looks like I got it right. Also squashed that typo.

**Pull request merged**